main.py

```diff
@@ -215,31 +166,31 @@ def processClips(urls, currentDirectory):
 
         downloadVideo(videoDirectory, videoId, url)
         comments = getComments(service, videoId)
-
-        timestamps = getTimestamps(comments)
-        for timestamp in timestamps:
-            clips.append(VideoFileClip(videoDirectory + "/" + videoId + ".mp4").subclip(timestamp[0], timestamp[1]))
-
-        slowMotionTimeStamps = getSlowMotionTimestamps(comments)
-        for timestamp in slowMotionTimeStamps:
-            clips.append(VideoFileClip(videoDirectory + "/" + videoId + ".mp4").subclip(timestamp[0][0], timestamp[0][1]))
-            clips.append(VideoFileClip(videoDirectory + "/" + videoId + ".mp4").subclip(timestamp[1][0], timestamp[1][1]).fx(vfx.speedx, 0.3))
-
-        downloadTimeStamps = getDownloadTimestamps(comments)
+        timestamps = getAllTimestamps(comments)
+        clipPath = videoDirectory + "/" + videoId + ".mp4"
         downloadsDirectory = currentDirectory + "/DownloadedClips"
-        if not os.path.exists(downloadsDirectory):
-            os.mkdir(downloadsDirectory)
-        for timestamp in downloadTimeStamps:
-            downloadClip = VideoFileClip(videoDirectory + "/" + videoId + ".mp4").subclip(timestamp[0], timestamp[1])
-            downloadClip.write_videofile(downloadsDirectory + "/" + str(timestamp[0]) + str(timestamp[1]) + ".mp4")
-            downloadClip.close()
-
-        timelapseTimeStamps = getTimeLapseTimestamps(comments)
-        for timestamp in timelapseTimeStamps:
-            new_clip = VideoFileClip(videoDirectory + "/" + videoId + ".mp4").subclip(timestamp[0], timestamp[1])
-            new_clip.audio = None
-            clips.append(new_clip.fx(vfx.speedx, 60))
-
+        
+        for timestamp in timestamps:
+            if timestamp.command == CLIP:
+                clips.append(VideoFileClip(clipPath).subclip(timestamp.startTime, timestamp.endTime))
+            elif timestamp.command == CLOSED_CAPTIONING:
+                # TODO
+                pass
+            elif timestamp.command == DOWNLOAD:
+                if not os.path.exists(downloadsDirectory):
+                    os.mkdir(downloadsDirectory)
+                downloadClip = VideoFileClip(clipPath).subclip(timestamp.startTime, timestamp.endTime)
+                downloadClip.write_videofile(downloadsDirectory + "/" + str(timestamp.startTime) + str(timestamp.endTime) + ".mp4")
+                downloadClip.close()
+            elif timestamp.command == FAST_FORWARD:
+                new_clip = VideoFileClip(clipPath).subclip(timestamp.startTime, timestamp.endTime)
+                new_clip.audio = None
+                clips.append(new_clip.fx(vfx.speedx, 60))
+            elif timestamp.command == NO_MUSIC:
+                # TODO
+                pass
+            elif timestamp.command == SLOW:
+                clips.append(VideoFileClip(clipPath).subclip(timestamp.startTime, timestamp.endTime).fx(vfx.speedx, 0.3))
```

Comment:
can we make a helper for `VideoFileClip(clipPath).subclip(timestamp.startTime, timestamp.endTime)` 

not a blocking issue